lib/open_food_network/scope_variant_to_hub.rb

```diff
@@ -29,21 +29,15 @@ def in_stock?
       end
 
       def count_on_hand
-        @variant_override.andand.count_on_hand || super
+        return super unless @variant_override.andand.stock_overridden?
+
+        @variant_override.count_on_hand
       end
 
       def on_demand
-        if @variant_override.andand.on_demand.nil?
-          if @variant_override.andand.count_on_hand.present?
-            # If we're overriding the stock level of an on_demand variant, show it as not
-            # on_demand, so our stock control can take effect.
-            false
-          else
-            super
-          end
-        else
-          @variant_override.andand.on_demand
-        end
+        return super if @variant_override.andand.on_demand.nil?
```

Comment:
I'm wondering if we should introduce an alias here:
```rb
def use_producer_settings?
  on_demand.nil?
end
```

It could also be:
```rb
def active?
  !on_demand.nil?
end
```

Not sure.

Reply:
yeah, I like it but I am not sure. I ended up following the first option.